File: Software Technologies/10. Java Basic Web/Calculator/src/main/java/com/softuni/controller/HomeController.java
package com.softuni.controller;

import com.softuni.entity.Calculator;
import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestParam;

@Controller
public class HomeController {
	@GetMapping("/")
	public String index(Model model) {
		model.addAttribute("operator", "+");
		model.addAttribute("view", "views/calculatorForm");
		return "base-layout";
	}

	@PostMapping("/")
	public String calculate(@RequestParam String leftOperand,
							@RequestParam String operator,
							@RequestParam String rightOperand,
							Model model) {

		double num1;
		double num2;

		boolean incorrectNumber = false;
		try {
		num1 = Double.parseDouble(leftOperand);
		} catch (NumberFormatException ex) {
			num1 = 0;
			incorrectNumber = true;
		}

		try {
			num2 = Double.parseDouble(rightOperand);
		} catch (NumberFormatException ex) {
			num2 = 0;
			incorrectNumber = true;
		}

		if(!incorrectNumber) {
			Calculator calculator = new Calculator(num1, num2, operator);

			double result = calculator.caluclateResult();

			model.addAttribute("leftOperand", calculator.getLeftOperand());
			model.addAttribute("operator", calculator.getOperator());
			model.addAttribute("rightOperand", calculator.getRightOperand());

			model.addAttribute("result", result);
		} else {
			model.addAttribute("leftOperand", "");
			model.addAttribute("operator", "");
			model.addAttribute("rightOperand", "");

			model.addAttribute("result", "");
		}

		model.addAttribute("view", "views/calculatorForm");
		return "base-layout";
	}
}

File: Software Technologies/12. C# ASP.NET MVC Overview/Calculator/Calculator-CSharp/Controllers/HomeController.cs
using System.Web.Mvc;
using Calculator_CSharp.Models;
using System;

namespace Calculator_CSharp.Controllers
{
    public class HomeController : Controller
    {
        public ActionResult Index(Calculator calculator)
        {
            return View(calculator);
        }

        [HttpPost]
        public ActionResult Calculate(Calculator calculator)
        {
            calculator.Result = CalculateResult(calculator);

            return RedirectToAction("Index", calculator);
        }

        public decimal CalculateResult(Calculator calculator)
        {
            decimal result = 0m;

            switch(calculator.Operator)
            {
                case "+":
                    result = calculator.LeftOperand + calculator.RightOperand;
                    break;

                case "-":
                    result = calculator.LeftOperand - calculator.RightOperand;
                    break;

                case "*":
                    result = calculator.LeftOperand * calculator.RightOperand;
                    break;

                case "/":
                    result = calculator.LeftOperand / calculator.RightOperand;
                    break;

                case "^":
                    result = (decimal)Math.Pow((double)calculator.LeftOperand, (double)calculator.RightOperand);
                    break;

                case "AND":
                    result = (long)calculator.LeftOperand & (long)calculator.RightOperand;
                    break;

                case "OR":
                    result = (long)calculator.LeftOperand | (long)calculator.RightOperand;
                    break;

                case "XOR":
                    result = (long)calculator.LeftOperand ^ (long)calculator.RightOperand;
                    break;
            }

            return result;
        }
    }
}